### PR TITLE
MULE-11286: Fix: avoid verification of timestamp action matching in request and response of a soap service.

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

File: modules/ws/src/main/java/org/mule/module/ws/security/NoTimestampEnforcerWSS4JInInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        for (Integer action : actions)
+        {
+            recordedActions.add(action);
+        }
+
+        for (WSSecurityEngineResult result : wsResult)
+        {
+            final Integer actInt = (Integer) result.get(WSSecurityEngineResult.TAG_ACTION);
+            int act = actInt.intValue();
+            if (act == WSConstants.SC || act == WSConstants.BST || act == WSConstants.TS)
```
Comment (elrodro83) [inline modules/ws/src/main/java/org/mule/module/ws/security/NoTimestampEnforcerWSS4JInInterceptor.java]: static imports.
Also, i don't think you need to call intValue() before

File: modules/ws/src/main/java/org/mule/module/ws/consumer/WSConsumer.java [hunk 2]
```diff
@@ -325,7 +326,7 @@ private CxfOutboundMessageProcessor createCxfOutboundMessageProcessor(WSSecurity
             }
             if (!inConfigProperties.isEmpty())
             {
-                cxfBuilder.getInInterceptors().add(new WSS4JInInterceptor(inConfigProperties));
+                cxfBuilder.getInInterceptors().add(new NoTimestampEnforcerWSS4JInInterceptor(inConfigProperties));
```
Comment (elrodro83) [inline modules/ws/src/main/java/org/mule/module/ws/consumer/WSConsumer.java]: would there be some scenario where we want to keep this validation anyway?

File: modules/ws/src/main/resources/META-INF/mule-ws.xsd [hunk 3]
```diff
@@ -241,6 +241,17 @@
                                                 </xsd:documentation>
                                             </xsd:annotation>
                                         </xsd:attribute>
+										<xsd:attribute name="checkResponseTimestamp" type="xsd:boolean"
```
Comment (elrodro83) [inline modules/ws/src/main/resources/META-INF/mule-ws.xsd]: ident with spaces

File: modules/ws/src/main/resources/META-INF/mule-ws.xsd [hunk 4]
```diff
@@ -241,6 +241,17 @@
...
+											use="optional" default="false">
+											<xsd:annotation>
+												<xsd:documentation>
+													If true, the difference between creation and expiry time
+													is verified in the response. You should set this attribute
+													to boolean if the server sends a timestamp as a response
```
Comment (elrodro83) [inline modules/ws/src/main/resources/META-INF/mule-ws.xsd]: to boolean -> to true

File: modules/ws/src/main/resources/META-INF/mule-ws.xsd [hunk 5]
```diff
@@ -241,6 +241,17 @@
                                                 </xsd:documentation>
                                             </xsd:annotation>
                                         </xsd:attribute>
+                                        <xsd:attribute name="checkResponseTimestamp" type="xsd:boolean"
+                                            use="optional" default="false">
+                                            <xsd:annotation>
+                                                <xsd:documentation>
+                                                    If true, the difference between creation and expiry time
+                                                    is verified in the response. You should set this attribute
+                                                    to boolean if the server sends a timestamp as a response
```
Comment (elrodro83) [inline modules/ws/src/main/resources/META-INF/mule-ws.xsd]: to boolean -> to true

File: modules/ws/src/test/java/org/mule/module/ws/functional/WssTimestampInResponseTestCase.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    @Override
+    protected String getConfigFile()
+    {
+        return "wss-timestamp-in-response-test-case.xml";
+    }
+
+    @Test
+    public void correctlyProcessTimestampInResponse() throws Exception
+    {
+
```
Comment (elrodro83) [inline modules/ws/src/test/java/org/mule/module/ws/functional/WssTimestampInResponseTestCase.java]: remove extra line

File: modules/ws/src/test/java/org/mule/module/ws/functional/WssTimestampInResponseTestCase.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+                                                      "</ns2:echoWithHeadersResponse>";
+
+    @Override
+    protected String getConfigFile()
+    {
+        return "wss-timestamp-in-response-test-case.xml";
+    }
+
+    @Test
+    public void correctlyProcessTimestampInResponse() throws Exception
```
Comment (elrodro83) [inline modules/ws/src/test/java/org/mule/module/ws/functional/WssTimestampInResponseTestCase.java]: change the name to reflect if this is validating the response timestamp or not. Add another test for the opposite case.

Comment (fsgonz) [inline modules/ws/src/test/java/org/mule/module/ws/functional/WssTimestampInResponseTestCase.java]: I have added tests for 4 scenarios:

TS in response and no TS action requested -> security error (that's the way it originally worked)
No TS in response and no TS action requested -> expected soap message as response (echo)
No TS in response and TS action requested -> security error
TS in response and TS action requested -> expected soap message as response (echo)

**Pull request merged**